akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala

```diff
@@ -108,6 +108,7 @@ private[http] class HttpResponseParser(protected val settings: ParserSettings, p
           maxResponseReasonLength + " characters")
       skipReason(startIdx)
     } else if (isNewLine(cursor + 3)) {
+      parseStatusCode()
```

Comment:
I'm not confident this does what you think, see below comment which explains that we actually DO accept such responses (and I remember the PR actually too..)